cmake/environment.cmake

```diff
@@ -9,23 +9,28 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     set(CMAKE_BUILD_TYPE "Release")
 endif()
 
-## Force default install prefix
-if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT OR NOT "${PROJECT_BINARY_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
-    set(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/install/${PROJECT_NAME}" CACHE PATH "Install path prefix, prepended onto install directories." FORCE)
-endif()
-message(STATUS "Install prefix: ${CMAKE_INSTALL_PREFIX}")
-
-## Set the output directories globally
-set(ARCHIVE_OUTPUT_DIRECTORY lib)
-set(RUNTIME_OUTPUT_DIRECTORY bin)
-if(WIN32)
-    set(LIBRARY_OUTPUT_DIRECTORY ${RUNTIME_OUTPUT_DIRECTORY})
-else()
-    set(LIBRARY_OUTPUT_DIRECTORY ${ARCHIVE_OUTPUT_DIRECTORY})
+## only for out-of-tree
+if(NOT TARGET Sofa.Config)
```

Comment:
Note for the future @fredroy : to test if we are in-tree, prefer `if(CMAKE_PROJECT_NAME STREQUAL "Sofa")`